Nine: migrate IRC to 'data' style

jaredgrubb opened the pull request:
This is still a bit of a WIP, but I wanted to get comments. I think it would actually work as-is, but some parts of it are not implemented (ETA, Builder Tags/Categories)

jaredgrubb committed:
Nine: migrate IRC to 'data' style

Jc2k committed:
Update release notes

djmitche commented on master/buildbot/status/words.py:
```diff
@@ -187,29 +187,57 @@ def doSilly(self, message):
+    def getOnlineBuilders(self):
+        all_buildslaves = yield self.master.data.get(('buildslaves',))
+        online_builderids = set()
+        for buildslave in all_buildslaves:
```
You probably want to skip buildslaves with `connected_to==[]`.  There's no way to write a ResultSpec filter for that, though, so it should just be a condition in the for loop.

djmitche commented on master/buildbot/status/words.py:
```diff
@@ -354,41 +396,71 @@ def command_NOTIFY(self, args, who):
+        d = self.master.data.get(('builds',),
+                        filters=[resultspec.Filter('builderid', 'eq', [builderid]),
+                                 resultspec.Filter('complete', 'eq', [True])],
+                        order=['number'],
```
You probably want `-number` here, to get the highest-numbered build?

andrewjcg committed:
Change semantics of `put` method for LRUCache objects

This changes the `LRUCache.put` method to provide a general
way to insert an object into the cache.  Previously,
`LRUCache.put` wasn't sufficient for this as it appeared to only
behave as a "touch" method (e.g. if the item wasn't already
in the cache, it was a noop).

This is useful to provide a way to "seed" the various caches.
In a number of locations we construct items and put them in
the DB, only to immediately load them back.  We already have
caches around these DB accesses to avoid multiple "gets"
actually hitting the DB, but we can avoid the initial "get"
missing in many cases by seeding the associated cache when
we put the item into the DB.

An example of this is chrdicts and the chrdict cache, where
it's common to call `addChange` followed by `getChange`.
This results in two DB accesses when it really only needs to
be one.

ludovicchabant committed:
Add more parameters to the Robocopy step on Windows.

ludovicchabant committed:
Added more tests for Robocopy.

djmitche committed:
Merge andrewjcg/buildbot:cache_set (PR #1125)

djmitche committed:
Merge ludovicchabant/buildbot:mswinsteps (PR #1129)

+autopep8, pyflakes

Commit:
For timeouts include the command being run in the failure string

Commit:
Remove duplicate in doc for 'all' email status mode.

alagoutte committed:
Add a master git+Gerrit configuration sample

djmitche committed:
ensure that summaryUnsubscribe succeeds

`summaryUnsubscribe` shoud be callable even if `summarySubscribe` hasn't
been called.  Fixes #2756.

rjogrady committed:
Support for --depth flag to repo init.

This allows checking out a shallow clone with reduced history.

Commit:
(fixes #2754)

Commit:
(fixes #2754)

djmitche committed:
Merge samuelmz/buildbot:fixes#2754 (PR #1134)

djmitche committed:
Merge edmorley/buildbot:master (PR #1130)

djmitche committed:
Merge dankegel/buildbot-x:master (PR #1131)

djmitche committed:
Merge alagoutte/buildbot:add_git_gerrit_example (PR #1132)

rjogrady committed:
Update docs and release notes for Repo --depth

djmitche committed:
Merge jaredgrubb/buildbot:builder-tags (PR #1103)

+autopep8

djmitche committed:
Merge rjogrady/buildbot:master (PR #1133)

djmitche committed:
rename repoDepth to just plain depth, and add a test for it

djmitche committed:
Merge branch 'master' into nine

Conflicts:
	master/buildbot/test/unit/test_master.py

djmitche committed:
replace vacuuous mock.assert_called with mock.assert_called_with()

mmartalo committed:
Fix #2735 pep8 > 1.4.6 has more errors

Jc2k committed:
Sort/group feature release notes

For example, all the Git source step notes are together.

Jc2k committed:
Fix formatting of :bb: macros in relnotes

Jc2k committed:
:bb: -> :bb:bug:

Jc2k committed:
Add missing bb:step and bb:sched

Jc2k committed:
More release note formatting

Jc2k committed:
Fix use of :py:mod:

Jc2k committed:
Move slave-side steps note to deprecations.

Jc2k committed:
Update relnotes with explicit versions

Jc2k committed:
archive release notes

Jc2k committed:
Don't send branch as ref/heads/master

Jc2k committed:
Add relnote for #1069

Jc2k committed:
Fix GitPoller setting invalid branch (PR #1069)

Jc2k committed:
Merge branch 'buildbot-0.8.9'

djmitche committed:
Skip flaky tests, with logging

This fixes #2748 (and opens a bunch of other bugs!)

djmitche committed:
don't try to fix PB on old Twisted versions

djmitche committed:
Can't use inline conditionals on Python-2.4

djmitche committed:
replace with statements with explicit patching

djmitche committed:
work around missing os.errno on py24

djmitche committed:
replicate any() for python-2.4

djmitche committed:
Merge branch 'py24-slave-compat' into buildbot-0.8.9

Fixes #2763

djmitche committed:
Merge branch 'buildbot-0.8.9'

djmitche committed:
Fix heading levels in developer/config.rst

Fixes #2736

djmitche committed:
add :contents: directives to config docs

This fixes #2737 and makes these docs easier to navigate.

djmitche committed:
Merge branch 'buildbot-0.8.9'

djmitche committed:
Merge branch 'master' into nine

Conflicts:
	master/buildbot/changes/gitpoller.py
	master/buildbot/test/integration/test_try_client.py
	master/buildbot/test/unit/test_changes_gitpoller.py
	master/docs/relnotes/index.rst

djmitche committed:
Merge mmartalo/buildbot:2735 (PR #1137)

+autopep8 (?!)

djmitche committed:
Merge pep8 changes into nine, plus more

This ignores E123 because pep8 wants indentation to look like

     def test_SourcedPropertiesValidator(self):
         self.doValidationTest(validation.SourcedPropertiesValidator(),
                               good=[
                                   {u'pname': ('{"a":"b"}', u'test')},
         ], bad=[
                                   None, 1, "hi",
                                   {u'pname': {'a': 'b'}},  # no source
                                   # name not unicode
                                   {'pname': ({'a': 'b'}, u'test')},
                                   # source not unicode
                                   {u'pname': ({'a': 'b'}, 'test')},
                                   # self is not json-able
                                   {u'pname': (self, u'test')},
         ])

which is ugly and most likely a bug in pep8.

djmitche committed:
ignore E123 in autopep8, too

jaredgrubb committed:
Address review comments

jaredgrubb commented:
This PR is a bit old .. cleaning up my old ones :)
